src/python/pants/backend/python/tasks2/gather_sources.py

```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+# Copyright 2016 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import shutil
+
+from pex.interpreter import PythonInterpreter
+from pex.pex import PEX
+from pex.pex_builder import PEXBuilder
+
+from pants.backend.python.targets.python_target import PythonTarget
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.build_graph.resources import Resources
+from pants.invalidation.cache_manager import VersionedTargetSet
+from pants.task.task import Task
+
+
+class GatherSources(Task):
+  """Gather local Python sources.
+
+  Creates an (unzipped) PEX on disk containing the local Python sources.
+  This PEX can be merged with a requirements PEX to create a unified Python environment
+  for running the relevant python code.
+  """
+
+  PYTHON_SOURCES = 'python_sources'
+
+  @classmethod
+  def product_types(cls):
+    return [cls.PYTHON_SOURCES]
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    round_manager.require_data(PythonInterpreter)
+
+  def execute(self):
+    targets = self.context.targets(lambda tgt: isinstance(tgt, PythonTarget) or
```

Comment:
The `isinstance(tgt, (PythonTarget, Resources))` form still fits in <100 cols and leverages the lib.